openfl/interface/aggregation_functions/fed_bagging.py

```diff
@@ -0,0 +1,130 @@
+# Copyright 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+
+"""Federated Boostrap Aggregation for XGBoost module."""
+
+import json
+
+import numpy as np
+
+from openfl.interface.aggregation_functions.core import AggregationFunction
+
+
+def get_global_model(iterator, target_round):
+    """
+    Retrieves the global model for the specific round from an iterator.
+
+    Parameters:
+    iterator (iterable): An iterable containing items with 'tags' and 'round' keys.
+    target_round (int): The round number for which the global model is to be retrieved.
+
+    Returns:
+    np.ndarray: The numpy array representing the global model for the specified round.
+    """
+    for item in iterator:
+        # Items tagged with ('model',) are the global model of that round
+        if "tags" in item and item["tags"] == ("model",) and item["round"] == target_round:
+            return item["nparray"]
+    raise ValueError(f"No item found with tag 'model' and round {target_round}")
+
+
+def append_trees(global_model, local_trees):
+    """
+    Appends local trees to the global model.
+
+    Parameters:
+    global_model (dict): A dictionary representing the global model.
+    local_trees (list): A list of dictionaries representing the local trees to be appended to the global model.
+
+    Returns:
+    dict: The updated global model with the local trees appended.
+    """
+    num_global_trees = int(
+        global_model["learner"]["gradient_booster"]["model"]["gbtree_model_param"]["num_trees"]
+    )
+    num_local_trees = len(local_trees)
+
+    global_model["learner"]["gradient_booster"]["model"]["gbtree_model_param"]["num_trees"] = str(
+        num_global_trees + num_local_trees
+    )
+    global_model["learner"]["gradient_booster"]["model"]["iteration_indptr"].append(
+        num_global_trees + num_local_trees
+    )
+    for new_tree in range(num_local_trees):
+        local_trees[new_tree]["id"] = num_global_trees + new_tree
+        global_model["learner"]["gradient_booster"]["model"]["trees"].append(local_trees[new_tree])
+        global_model["learner"]["gradient_booster"]["model"]["tree_info"].append(0)
+
+    return global_model
+
+
+class FedBaggingXGBoost(AggregationFunction):
```

Comment:
When someone contributes an aggregation function, is there an expectation/facilities for unit testing it?

Reply:
This has been a question I've been musing on in the back of my mind for a while. I don't think there are unit tests in place and seems only testable via local E2E run, but I am wondering if there is a basic test a user can run with some dummy tensors to make sure that there are no bugs in the logic, since they presumably would know what the result would be (or have their own working implementation from their initial development). Open to suggestions though

Reply:
I've already listed this in the backlog (post 1.7, most likely). This could include some some sort of contributor docs that outlines the process - APIs, test facilities, examples, etc. Speaking of APIs, it seems we only support the `AggregationFunction` abstraction for TaskRunner API. We should probably introduce something similar for Workflow API too. Some of the newly joining team members are already refactoring the FedProx-related examples to Workflow API, so they may take this up at some point in the future. However your support will certainly be needed as you know the practical needs of external contributors of such algorithms.